src/hooks/usePlayList.js

```diff
@@ -1,87 +1,64 @@
+// @ts-check
 import { useEffect, useState } from 'react'
 import getPlaylistData from '../utils/getPlaylistData'
+import getWrongAnswers from '../utils/getWrongAnswers'
 import shuffle from '../utils/shuffleArray'
 
 export default function usePlayList(playlist) {
-  const newPlaylist = shuffle(playlist)
+  const newPlaylist = playlist
   const [playlistData, setPlaylistData] = useState(null)
   const [answers, setAnswers] = useState(null)
   const [counter, setCounter] = useState(null)
   const [wrongAnswers, setWrongAnswers] = useState(null)
   let getNextUrl
 
-  // useEffect(()=> {
-
-  // })
-
-  //get playlist data
+  // get playlist data
   useEffect(() => {
-    const baseUrl = '/api/playlist/'
-    Promise.all(
-      newPlaylist.map(({ id }) => fetch(baseUrl + id).then(res => res.json()))
-    ).then(data => {
-      const newPlaylistData = data.map(({ results }) => results[0])
-      const cleanPlaylistData = newPlaylistData.filter(el => !!el)
-      !playlistData && setPlaylistData(cleanPlaylistData)
-      setCounter(cleanPlaylistData.length - 1)
-    })
-  }, [])
+    async function getData() {
+      const data = await getPlaylistData(shuffle(newPlaylist))
+      if (data) {
+        setPlaylistData(data)
+        setCounter(data?.length - 1)
+      }
+    }
+    getData()
+  }, [newPlaylist])
 
-  //get the wrong answers
+  //get wrong answers
   useEffect(() => {
     if (!counter) return
-    const interpret = playlistData[counter].artistName
-    const baseUrl = `/api/artist/${interpret}`
-    fetch(baseUrl)
-      .then(res => res.json())
-      .then(data => {
-        const wrongSongTitle = data.results.filter(
-          ({ trackName, artistName }) => {
-            if (
-              trackName === 'Undefined' ||
-              trackName === '(Un)Defined' ||
-              playlistData[counter].artistName !== artistName
-            ) {
-              return ''
-            }
-            return trackName
-          }
-        )
-        const wrongsongTitleSet = new Set(
-          wrongSongTitle.map(({ trackName }) => trackName)
-        )
-        setWrongAnswers([...wrongsongTitleSet])
-      })
-      .catch(error => console.error(error))
-  }, [playlistData, counter])
+    async function getData() {
+      const data = await getWrongAnswers(playlistData[counter].artistName)
+      if (data) {
```

Comment:
`getData`, `data` - this is very generic as you clearly handle `wrongAnswers` here.